internal/controllers/common/resource_definitions/resource_definitions.go

```diff
@@ -691,6 +757,40 @@ func NewCoreContainer(cr *model.CryostatInstance, specs *ServiceSpecs, imageTag
 		},
 	}
 
+	optional := false
+	secretName := cr.Name + "-db-connection-key"
+	envs = append(envs, corev1.EnvVar{
+		Name: "QUARKUS_DATASOURCE_PASSWORD",
+		ValueFrom: &corev1.EnvVarSource{
+			SecretKeyRef: &corev1.SecretKeySelector{
+				LocalObjectReference: corev1.LocalObjectReference{
+					Name: secretName,
+				},
+				Key:      "CONNECTION_KEY",
+				Optional: &optional,
+			},
+		},
+	})
+
+	secretName = cr.Name + "-storage-secret-key"
```

Comment:
```
  Warning  Failed          11s   kubelet            Error: secret "cryostat-sample-storage-secret-key" not found
```

This ends up assuming that the `-storage-secret-key` secret already exists, but it doesn't look like it does. I think this should be a key that the Operator generates and stores in a Secret. Later (in a future release) we can make this configurable for the user via the CR, when we also open it up to allowing the user to configure some other S3 storage instance.

Same goes for the `-db-connection-key` above.

There is already precedent for this from 2.4: https://github.com/cryostatio/cryostat-operator/blob/6b63e3f67a6ac6dda5ac307dc5d0a1adea444ec8/internal/controllers/secrets.go#L116